test/common/http/filter/ip_tagging_filter_test.cc

```diff
@@ -65,32 +49,234 @@ class IpTaggingFilterTest : public testing::Test {
+  EXPECT_CALL(stats_, counter("prefix.ip_tagging.total")).Times(1);
+
+  EXPECT_EQ(FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, false));
+  EXPECT_EQ("test_internal", request_headers.get_(Headers::get().EnvoyIpTags));
+
+  EXPECT_EQ(FilterDataStatus::Continue, filter_->decodeData(data_, false));
+  EXPECT_EQ(FilterTrailersStatus::Continue, filter_->decodeTrailers(request_headers));
+}
+
 TEST_F(IpTaggingFilterTest, InternalRequest) {
-  SetUpTest(internal_request_json);
+  SetUpTest(internal_request_yaml);
+  EXPECT_EQ(FilterRequestType::INTERNAL, config_->requestType());
+  TestHeaderMapImpl request_headers{{"x-envoy-internal", "true"}};
+
+  Network::Address::InstanceConstSharedPtr remote_address =
+      Network::Utility::parseInternetAddress("1.2.3.5");
+  EXPECT_CALL(filter_callbacks_.request_info_, downstreamRemoteAddress())
+      .WillOnce(ReturnRef(remote_address));
+
+  EXPECT_CALL(stats_, counter("prefix.ip_tagging.internal_request.hit")).Times(1);
+  EXPECT_CALL(stats_, counter("prefix.ip_tagging.total")).Times(1);
+
+  EXPECT_EQ(FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, false));
+  EXPECT_EQ("internal_request", request_headers.get_(Headers::get().EnvoyIpTags));
 
-  EXPECT_EQ(FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers_, false));
   EXPECT_EQ(FilterDataStatus::Continue, filter_->decodeData(data_, false));
-  EXPECT_EQ(FilterTrailersStatus::Continue, filter_->decodeTrailers(request_headers_));
+  EXPECT_EQ(FilterTrailersStatus::Continue, filter_->decodeTrailers(request_headers));
 }
 
 TEST_F(IpTaggingFilterTest, ExternalRequest) {
-  SetUpTest(external_request_json);
+  std::string external_request_yaml = R"EOF(
+request_type: external
+ip_tags:
+  - ip_tag_name: external_request
+    ip_list:
+      - {address_prefix: 1.2.3.4, prefix_len: 32}
+)EOF";
+  SetUpTest(external_request_yaml);
+  EXPECT_EQ(FilterRequestType::EXTERNAL, config_->requestType());
+  TestHeaderMapImpl request_headers;
+
+  EXPECT_CALL(stats_, counter("prefix.ip_tagging.total")).Times(1);
+  EXPECT_CALL(stats_, counter("prefix.ip_tagging.external_request.hit")).Times(1);
+
+  Network::Address::InstanceConstSharedPtr remote_address =
+      Network::Utility::parseInternetAddress("1.2.3.4");
+  EXPECT_CALL(filter_callbacks_.request_info_, downstreamRemoteAddress())
+      .WillOnce(ReturnRef(remote_address));
+
+  EXPECT_EQ(FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, false));
+  EXPECT_EQ("external_request", request_headers.get_(Headers::get().EnvoyIpTags));
 
-  EXPECT_EQ(FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers_, false));
   EXPECT_EQ(FilterDataStatus::Continue, filter_->decodeData(data_, false));
-  EXPECT_EQ(FilterTrailersStatus::Continue, filter_->decodeTrailers(request_headers_));
+  EXPECT_EQ(FilterTrailersStatus::Continue, filter_->decodeTrailers(request_headers));
 }
```

Comment:
I'd encourage doing an internal request here and verifying it doesn't increment total hits

ditto external request on the internal test case